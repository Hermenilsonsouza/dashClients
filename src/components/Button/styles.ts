import styled, { css } from 'styled-components'
import { ButtonProps } from '.'

const WrapperModifies = {
  primary: () => css`
    background-color: #E07b28;
    color: #FFF;
    &:hover {
      background-color: #FFF;
      border: 1px solid #E07b28;
      color: #E07b28;
    }
  `,

  secondary: () => css`
    background-color: #FFF;
    border: 1px solid #E07b28;
    color: #E07b28;

    &:hover {
      background-color: #E07b28;
      color: #FFF;
    }
  `
}

type ButtonStyleProps = Pick<ButtonProps, 'buttontype' >

export const Button = styled.button<ButtonStyleProps>`

  ${({buttontype}) => css`
    font-size: 16px;
    padding: 8px 16px;
    border-radius: 4px;
    box-shadow: none;
    border: 0;
    cursor: pointer;

    ${!!buttontype && WrapperModifies[buttontype]}
  `}


`
