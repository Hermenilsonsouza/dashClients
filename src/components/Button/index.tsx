import { AnchorHTMLAttributes, ButtonHTMLAttributes } from 'react'
import * as S from './styles'

type ButtonTypes =
  | AnchorHTMLAttributes<HTMLAnchorElement>
  | ButtonHTMLAttributes<HTMLButtonElement>

export type ButtonProps = {
  title: string
  buttontype: 'primary' | 'secondary'
} & ButtonTypes

const Button = ({ title, buttontype }: ButtonProps) => (
  <S.Button buttontype={buttontype}>{title}</S.Button>
)

export default Button
