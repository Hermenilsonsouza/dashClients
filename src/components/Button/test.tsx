import { render, screen } from '@testing-library/react'

import Button from '.'

describe('<Button />', () => {
  it('should render the heading', () => {
    const { container } = render(<Button title="editar" buttontype="primary" />)

    expect(screen.getByText('editar')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  }),
    it('should render button with spercif title', () => {
      render(<Button title="editar" buttontype="primary" />)
      expect(
        screen.getByRole('button', { name: /editar/i })
      ).toBeInTheDocument()
    }),
    it('should render with white background', () => {
      render(<Button title="editar" buttontype="secondary" />)

      expect(screen.getByRole('button', { name: /editar/i })).toHaveStyle({
        'background-color': '#FFF',
        color: '#E07b28'
      })
    }),
    it('should render with orange background', () => {
      render(<Button title="editar" buttontype="primary" />)

      expect(screen.getByRole('button', { name: /editar/i })).toHaveStyle({
        'background-color': '#E07b28',
        color: '#FFF'
      })
    })
})
