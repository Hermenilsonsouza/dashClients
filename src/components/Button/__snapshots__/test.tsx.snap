// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the heading 1`] = `
.c0 {
  font-size: 16px;
  padding: 8px 16px;
  border-radius: 4px;
  box-shadow: none;
  border: 0;
  cursor: pointer;
  background-color: #E07b28;
  color: #FFF;
}

.c0:hover {
  background-color: #FFF;
  border: 1px solid #E07b28;
  color: #E07b28;
}

<button
  buttontype="primary"
  class="c0"
>
  editar
</button>
`;
